### PR TITLE
Multi region read/write

### PR DESCRIPTION
**Pull request by ricmrodrigues**
adding the option to enable multi region read/write with the region provided in the factory ctor

### CODE REVIEW

File: src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs [hunk 1]
```diff
@@ -7,7 +7,9 @@ public class CosmosCacheFactorySettings
         public int DefaultTimeToLive { get; set; } = -1;  //never expire by default
         public bool UseKeyAsPartitionKey {get;set;} = false;
         public CosmosCacheFactoryIndexingSettings IndexingSettings { get; set; } = new CosmosCacheFactoryIndexingSettings();
+        public bool MultiRegionReadWrite { get; set; } = false;
```
Comment (arturek) [inline src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs]: assigning the default value is not necessary (here and in the other places).

File: src/Eshopworld.Caching.Cosmos/Eshopworld.Caching.Cosmos.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
 
-    <Version>1.3.0</Version>
+    <Version>1.5.0</Version>
```
Comment (arturek) [inline src/Eshopworld.Caching.Cosmos/Eshopworld.Caching.Cosmos.csproj]: what happened with 4? I'm just curious.

File: src/Eshopworld.Caching.Cosmos/CosmosCacheFactory.cs [hunk 3]
```diff
@@ -52,6 +51,30 @@ protected virtual ICache<T> BuildCacheInstance<T>(Uri documentCollectionUri)
             return new CosmosCache<T>(documentCollectionUri, DocumentClient, _settings.InsertMode, _settings.UseKeyAsPartitionKey);
         }
 
+        private ConnectionPolicy GetConnectionPolicy(CosmosCacheFactorySettings settings)
```
Comment (arturek) [inline src/Eshopworld.Caching.Cosmos/CosmosCacheFactory.cs]: please make it static

File: src/Eshopworld.Caching.Cosmos/CosmosCacheFactory.cs [hunk 4]
```diff
@@ -29,12 +29,11 @@ public CosmosCacheFactory(Uri cosmosAccountEndpoint, string cosmosAccountKey, st
             _dbName = dbName ?? throw new ArgumentNullException(nameof(dbName));
             _settings = settings;
```
Comment (arturek) [inline src/Eshopworld.Caching.Cosmos/CosmosCacheFactory.cs]: check for null value needed

File: src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs (hunk 1 above)
Comment (ricmrodrigues) [reply src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs]: I know, was just keeping it consistent with the pre-existing code style, OSS etiquette :)

File: src/Eshopworld.Caching.Cosmos/Eshopworld.Caching.Cosmos.csproj (hunk 2 above)
Comment (ricmrodrigues) [reply src/Eshopworld.Caching.Cosmos/Eshopworld.Caching.Cosmos.csproj]: it's the other PR, I know it creates an order issue, can fix if this one goes first

File: src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs (hunk 1 above)
Comment (arturek) [reply src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs]: I can see only one earlier example of such code (UseKeyAsPartitionKey). And this PR adds two more properties with default initializers :-) .

Comment (ricmrodrigues) [reply src/Eshopworld.Caching.Cosmos/CosmosCacheFactorySettings.cs]: I can fix them all, no bother :-)

File: src/Eshopworld.Caching.Cosmos/CosmosCacheFactory.cs (hunk 3 above)
Comment (ricmrodrigues) [reply src/Eshopworld.Caching.Cosmos/CosmosCacheFactory.cs]: done

**Pull request merged**